samples/mixed/consumer/src/main.rs

```diff
@@ -0,0 +1,283 @@
+
+            let client_result = ProviderClient::connect(provider_uri.clone()).await;
+            if client_result.is_err() {
+                continue;
+            }
+            let mut client = client_result.unwrap();
+
+            let response_id = Uuid::new_v4().to_string();
+
+            let payload: String = format!("{is_active}");
+
+            let request = tonic::Request::new(InvokeRequest {
+                entity_id: String::from(ACTIVATE_AIR_CONDITIOING_COMMAND_ID),
+                consumer_uri: consumer_uri.clone(),
+                response_id,
+                payload,
+            });
+
+            let _response = client.invoke(request).await;
+
+            is_active = !is_active;
+
+            sleep(Duration::from_secs(20)).await;
+        }
+    });
+}
+
+async fn get_provider_uri(entity_id: &str) -> Result<String, Box<dyn std::error::Error>> {
+    // Obtain the DTDL for the send_notification command.
+    info!("Sending a find_by_id request to the Digital Twin Service for the DTDL for the send_notification command.");
+    let mut client = DigitalTwinClient::connect("http://[::1]:50010").await?; // Devskim: ignore DS137138
+    let request = tonic::Request::new(FindByIdRequest { entity_id: String::from(entity_id) });
+    let response = client.find_by_id(request).await?;
+    let dtdl = response.into_inner().dtdl;
+    info!("Received the response for the find_by_id request. The DTDL is:\n{}", &dtdl);
+
+    info!("Parsing the DTDL.");
+    let mut parser = ModelParser::new();
+    let json_texts = vec![dtdl];
+    let model_dict_result = parser.parse(&json_texts);
+    if let Err(error) = model_dict_result {
+        panic!("Failed to parse the DTDL: {error}");
+    }
+    let model_dict = model_dict_result.unwrap();
+    info!("The DTDL parser has successfully parsed the DTDL.");
+
+    // Create the id (as a DTMI) for the send_notification command.
+    let send_notification_command_id: Option<Dtmi> = create_dtmi(entity_id);
+    if send_notification_command_id.is_none() {
+        panic!("Unable to create the dtmi");
+    }
+
+    // Get the entity from the DTDL for the send notification command.
+    let entity_result = model_dict.get(&send_notification_command_id.unwrap());
+    if entity_result.is_none() {
+        panic!("Unable to find the entity");
+    }
+    let entity = entity_result.unwrap();
+
+    // Get the URI property from the entity.
+    let uri_property_result = entity.undefined_properties().get(URI_PROPERTY_ID);
+    if uri_property_result.is_none() {
+        panic!("Unable to find the URI property");
+    }
+    let uri_property = uri_property_result.unwrap();
+
+    // Get the value for the URI property.
+    let uri_property_value_result = uri_property.get("@value");
+    if uri_property_value_result.is_none() {
+        info!("Unable to find the value for the URI for the ambient air temperature's provider.");
+    }
+    let uri_property_value = uri_property_value_result.unwrap();
```

Comment:
what happens here if uri_property_value_result.is_none() is true?

Reply:
Good catch!  Fixed.